From 26be71ad32cb5edd4c2bf5b45e92e2ae664eb957 Mon Sep 17 00:00:00 2001
From: Javier Miranda <miranda@adacore.com>
Date: Sat, 4 Oct 2025 07:06:34 +0000
Subject: [PATCH] Fix sources after enforcing RM 8.6(27.1) in the compiler

* gtkada-canvas_view.ads (Inner_Most_Item): Add explicit type conversion.

For eng/toolchain/gnat#1292
---
 src/gtkada-canvas_view.ads | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/gtkada-canvas_view.ads b/src/gtkada-canvas_view.ads
index 6038fe8cc..7ad5321af 100644
--- a/src/gtkada-canvas_view.ads
+++ b/src/gtkada-canvas_view.ads
@@ -589,7 +589,7 @@ package Gtkada.Canvas_View is
      (Self           : not null access Canvas_Item_Record;
       Dummy_At_Point : Model_Point;
       Dummy_Context  : Draw_Context)
-      return Abstract_Item is (Self);
+      return Abstract_Item is (Abstract_Item (Self));
    overriding function Position
      (Self : not null access Canvas_Item_Record) return Gtkada.Style.Point;
    overriding function Contains
@@ -2027,7 +2027,7 @@ package Gtkada.Canvas_View is
      (Self           : not null access Canvas_Link_Record;
       Dummy_At_Point : Model_Point;
       Dummy_Context  : Draw_Context)
-      return Abstract_Item is (Self);
+      return Abstract_Item is (Abstract_Item (Self));
    overriding function Parent
      (Self : not null access Canvas_Link_Record)
       return Abstract_Item is (null);
